### DISCUSSION TITLE
Environment with id '4' already contains a virtual machine resource with name 'xyz'

### DISCUSSION THREAD

**Discussion by dazinator** [microsoft/azure-pipelines-agent#4225] [Q&A]
I was testing deployment of a dockerised agent (linux)
It ran fine and appeared in DevOps (Pipelines > Environments > VM Resource)
I then deleted the VM, re-created a new VM, and re-deployed the agent to the VM running as a docker container again.
The VM resource was never removed from DevOps.
However I can no longer deploy to the agent and checking its logs I see:
```
Environment with id '4' already contains a virtual machine resource with name 'xyz'.
```

Is there any way to make this agent automatically replace an agent with an existing name in devops so that is doesn't fail on startup?

**Comment by dazinator** [comment]
still no news?

**Reply by acottuli** [reply]
@dazinator I just ran into the same issue. Passing the flag `--replace` to `./config.sh` fixed the issue for me. 

From the output of `./config.sh --help`

```
Configure options:
 --pool <pool>                     Pool name for the agent to join
 --agent <agent>                   Agent name
 --replace                         Replace the agent in a pool. If another agent is listening by that
                                   name, it will start failing with a conflict.
 --work <workDirectory>            Work directory where job data is stored. Defaults to _work under the
                                   root of the agent directory. The work directory is owned by a given
                                   agent and should not share between multiple agents.
 --acceptTeeEula                   macOS and Linux only. Accept the TEE end user license agreement.
 --gitUseSChannel                  Windows only. Tell Git to use Windows' native cert store.
 --alwaysExtractTask               Perform an unzip for tasks for each pipeline step.
```

If you want to dig deeper, the relevant source code can be found here https://github.com/microsoft/azure-pipelines-agent/blob/e5dca166f045c18e234eeae3ba1556a58fabe884/src/Agent.Listener/Configuration/ConfigurationManager.cs#L219-L245
